modules/table/src/main/java/org/apache/ignite/internal/table/distributed/storage/InternalTableImpl.java

```diff
@@ -1485,44 +1468,66 @@ public int partitionId(BinaryRowEx row) {
     }
 
     /**
-     * Gathers the result of batch processing into a single resulting collection of rows.
+     * Gets a batch result.
      *
      * @param rowBatches Row batches.
      * @return Future of collecting results.
      */
-    static CompletableFuture<Collection<BinaryRow>> collectMultiRowsResponsesWithoutRestoreOrder(Collection<RowBatch> rowBatches) {
-        return allResultFutures(rowBatches)
-                .thenApply(response -> {
-                    var result = new ArrayList<BinaryRow>(rowBatches.size());
-
-                    for (RowBatch rowBatch : rowBatches) {
-                        Collection<BinaryRow> batchResult = (Collection<BinaryRow>) rowBatch.getCompletedResult();
+    public static CompletableFuture<List<BinaryRow>> collectRejectedRowsResponsesWithRestoreOrder(Collection<RowBatch> rowBatches) {
+        return collectMultiRowsResponsesWithRestoreOrder(
+                rowBatches,
+                batch -> {
+                    List<BinaryRow> result = new ArrayList<>();
+                    List<Boolean> response = (List<Boolean>) batch.getCompletedResult();
```

Comment:
`java.util.BitSet` will be more effective.

Reply:
The same as I already explain above.

Reply:
It turns out that you are transfer a list of Boolean objects over the network, instead of a simple `BitSet`, this is bad.

Reply:
It might be bad, but it already was. Probably, we will be able to optimize it in the future.
But the colletion of boolean is better that collection of Binary rowa waht that have been for part of batch operations (insertAll, deleteAllExact) befo my patch.

Reply:
Moreover, for the last change in the merge we are using other approach to transfer result for batch operations.

Reply:
Fixing it to `BitSet` shouldn't be a problem, it's the right thing to do since we don't know when we'll fix it.